legion_core/src/world.rs

```diff
@@ -712,6 +717,56 @@ impl World {
         unsafe { self.get_component_mut_unchecked(entity) }
     }
 
+    pub fn get_all_components<'a, T: Component>(&'a self) -> RefMapSet<'a, Vec<&'a T>> {
+        let filter = crate::filter::filter_fns::component::<T>();
+
+        let mut borrows = vec![];
+        let mut refs = vec![];
+
+        unsafe {
+            filter
+                .iter_archetype_indexes(self.storage())
+                .flat_map(|archetype_index| {
+                    self.storage()
+                        .archetypes()
+                        .get_unchecked(archetype_index.0)
+                        .iter_data_slice::<T>()
+                })
+                .map(|x| x.deconstruct())
+                .for_each(|(borrow, slice)| {
+                    borrows.push(borrow);
+                    refs.extend(slice);
+                });
+        }
+
+        RefMapSet::new(borrows, refs)
+    }
+
+    pub fn get_all_components_mut<'a, T: Component>(&'a self) -> RefMapMutSet<'a, Vec<&'a mut T>> {
```

Comment:
doc string

Reply:
Same as above

Reply:
This PR includes some changes that are redundant with #105, I'm hoping we can get that one in first and rebase this on top of it. #105 has doc strings for these functions.